### PR TITLE
Added text.base64Encode and text.base64Decode

### PR DESCRIPTION
**Pull request by mpilquist**
Adapted the implementations from scodec-bits for streaming / continuation.

### CODE REVIEW

**Comment by mpilquist** [comment]
Fixes #1587

File: core/shared/src/main/scala/fs2/text.scala [hunk 1]
```diff
@@ -201,4 +204,219 @@ object text {
...
+                    s"Unexpected character '$c' at index $idx after padding character; only '=' and whitespace characters allowed after first padding character"
+                  )
+                }
+              }
+            }
+            mod match {
+              case 0 =>
+                buffer = (cidx & 0x3f)
+                mod += 1
+              case 1 =>
```
Comment (matthughes) [inline core/shared/src/main/scala/fs2/text.scala]: `case 1 | 2` might make intent a little clearer

File: core/shared/src/main/scala/fs2/text.scala [hunk 2]
```diff
@@ -201,4 +204,219 @@ object text {
...
+    }
+
+    def finish(state: State): Either[String, Chunk[Byte]] =
+      if (state.padding != 0 && state.mod != 0) paddingError
+      else
+        state.mod match {
+          case 0 => Right(Chunk.empty)
+          case 1 => Left("Final base 64 quantum had only 1 digit - must have at least 2 digits")
+          case 2 =>
+            Right(Chunk(((state.buffer >> 4) & 0x0ff).toByte))
```
Comment (matthughes) [inline core/shared/src/main/scala/fs2/text.scala]: Is there a reason this is `0x0ff` vs `0xff`?

File: core/shared/src/test/scala/fs2/TextSpec.scala [hunk 3]
```diff
@@ -230,5 +234,28 @@ class TextSpec extends Fs2Spec {
...
+          .through {
+            if (unchunked) _.unchunk
+            else _.rechunkRandomlyWithSeed(0.1, 2.0)(rechunkSeed)
+          }
+          .through(text.base64Decode[Fallible])
+          .compile
+          .to(ByteVector) shouldBe
+          Right(bs.map(ByteVector.view(_)).foldLeft(ByteVector.empty)(_ ++ _))
+      }
+    }
```
Comment (matthughes) [inline core/shared/src/test/scala/fs2/TextSpec.scala]: Maybe repeat the encode/decode test but add random whitespace at beginning and end?

File: core/shared/src/main/scala/fs2/text.scala [hunk 4]
```diff
@@ -201,4 +204,219 @@ object text {
...
+              val out = new String(
+                Array(
+                  alphabet.toChar(first),
+                  alphabet.toChar(second),
+                  alphabet.toChar(third),
+                  alphabet.pad
+                )
+              )
+              Pull.output1(out)
+            case _ => sys.error("carry must be size 0, 1, or 2")
```
Comment (matthughes) [inline core/shared/src/main/scala/fs2/text.scala]: Consider adding the invalid carry size in the error

File: core/shared/src/test/scala/fs2/TextSpec.scala (hunk 3 above)
Comment (matthughes) [inline core/shared/src/test/scala/fs2/TextSpec.scala]: Could also add test feeding the decode invalid base64 characters and assert it fails as expected (not throws exception).

Comment (matthughes) [reply core/shared/src/test/scala/fs2/TextSpec.scala]: Or invalid lengths.

File: core/shared/src/main/scala/fs2/text.scala (hunk 2 above)
Comment (mpilquist) [reply core/shared/src/main/scala/fs2/text.scala]: No, old habit. Fixing.

**Comment by mpilquist** [comment]
@matthughes Want to take a quick look at updates from review?

**Pull request merged**